packages/http-specs/specs/payload/pageable/main.tsp

```diff
@@ -52,4 +52,168 @@ namespace ServerDrivenPagination {
+      }
+      ```
+      
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-query-response-body?token=12345
+      
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+        ]
+      }
+      ```
+      """)
+    @route("/request-query-response-body")
+    @list
+    op reqQueryResBody(@continuationToken @query token?: string): {
+      @pageItems
+      pets: Pet[];
+
+      @continuationToken nextToken?: string;
+    };
+
+    @scenario
+    @scenarioDoc("""
+      Test case for using continuation token as pagination. Continuation token is passed in the request header and response body.
+      
+      Two requests need to be tested.
+      1. Initial request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-header-response-body      
+      
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "1", "name": "dog" },
+          { "id": "2", "name": "cat" }
+        ],
+        "nextToken": "12345"
+      }
+      ```
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-header-response-body
+      
+      Expected header:
+      token=12345
+      
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+        ]
+      }
+      ```
+      """)
+    @route("/request-header-response-body")
+    @list
+    op reqHeaderResBody(@continuationToken @header token?: string): {
+      @pageItems
+      pets: Pet[];
+
+      @continuationToken nextToken?: string;
+    };
+
+    @scenario
+    @scenarioDoc("""
+      Test case for using continuation token as pagination. Continuation token is passed in the request query and response header.
+      
+      Two requests need to be tested.
+      1. Initial request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-query-response-header
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "1", "name": "dog" },
+          { "id": "2", "name": "cat" }
+        ]
+      }
+      ```
+      
+      Expected response header:
+      next-token=12345
+      
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-query-response-header?token=12345
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+        ]
+      }
+      ```
+      """)
+    @route("/request-query-response-header")
+    @list
+    op reqQueryResHeader(@continuationToken @query token?: string): {
+      @pageItems
+      pets: Pet[];
+
+      @continuationToken @header nextToken?: string;
+    };
```

Comment:
@tadelesh tcgc 0.51.0 would have `resultPath=undefined` in this case

![{F4A32C14-8F50-4F3F-980E-CCF068260ECF}](https://github.com/user-attachments/assets/87100184-c036-443a-b8a9-52275e34304a)

Reply:
good catch. it is a bug in tcgc and fixed in this pr: https://github.com/Azure/typespec-azure/pull/2229